spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/Mqttv5ClientManager.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mqtt.core;
+
+import org.eclipse.paho.mqttv5.client.IMqttAsyncClient;
+import org.eclipse.paho.mqttv5.client.IMqttToken;
+import org.eclipse.paho.mqttv5.client.MqttAsyncClient;
+import org.eclipse.paho.mqttv5.client.MqttCallback;
+import org.eclipse.paho.mqttv5.client.MqttConnectionOptions;
+import org.eclipse.paho.mqttv5.client.MqttDisconnectResponse;
+import org.eclipse.paho.mqttv5.common.MqttException;
+import org.eclipse.paho.mqttv5.common.MqttMessage;
+import org.eclipse.paho.mqttv5.common.packet.MqttProperties;
+
+import org.springframework.util.Assert;
+
+public class Mqttv5ClientManager extends AbstractMqttClientManager<IMqttAsyncClient> implements MqttCallback {
+
+	private final MqttConnectionOptions connectionOptions;
+
+	private volatile IMqttAsyncClient client;
+
+	public Mqttv5ClientManager(String url, String clientId) {
+		super(clientId);
+		Assert.notNull(url, "'url' is required");
+		setUrl(url);
+		this.connectionOptions = new MqttConnectionOptions();
+		this.connectionOptions.setServerURIs(new String[]{ url });
+		this.connectionOptions.setAutomaticReconnect(true);
+	}
+
+	public Mqttv5ClientManager(MqttConnectionOptions connectionOptions, String clientId) {
+		super(clientId);
+		Assert.notNull(connectionOptions, "'connectionOptions' is required");
+		this.connectionOptions = connectionOptions;
+		if (!this.connectionOptions.isAutomaticReconnect()) {
+			logger.warn("It is recommended to set 'automaticReconnect' MQTT connection option. " +
+					"Otherwise connection check and reconnect should be done manually.");
```

Comment:
Now the problem with this option that it is really `optional` for the `Mqttv5PahoMessageHandler`.
So, it's OK to have this warn over here, but it has to be slightly modified:
```
"If this `ClientManager` is used from message-driven channel adapters, it is recommended to set 'automaticReconnect' MQTT connection option. " +
					"Otherwise connection check and reconnect should be done manually."
```
And then it is probably not warn, but `info`.
It does not break anything: just need to be aware of catching `MqttConnectionFailedEvent`s to reconnect manually. 
We say about that in the docs: https://docs.spring.io/spring-integration/docs/current/reference/html/mqtt.html#mqtt-v5

> Otherwise, only the manual restart of `Mqttv5PahoMessageDrivenChannelAdapter` can handle reconnects, e.g. via `MqttConnectionFailedEvent` handling on disconnection.